docs/client_retries.md

```diff
@@ -0,0 +1,320 @@
+# Client Side Retries
+Client Libraries use retries to handle unexpected, transient failures (i.e. server is temporarily unavailable).
+Multiple attempts, hopefully, will result in a successful response from the server.
+
+Default retry values are selected by the team operating the cloud service. These retry values are configured
+per RPC. A service *may* choose to only enable retries for a subset of RPCs. It is possible that each RPC for
+a service is configured differently.
+
+## Retry Parameters
+Client libraries have two types of retry parameters to configure:
+1. Retry Status Code: Set of failure status codes to retry on
+2. Retry Time/ Attempt Bounds: Configurable [RetrySettings](https://cloud.google.com/java/docs/reference/gax/latest/com.google.api.gax.retrying.RetrySettings) to define the bounds
+
+### RPC Retry Configurations
+Using Java-Asset v3.41.0 as an example, the default retry configurations are defined in the following places:
+
+Retry Status Codes are configured [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1058-L1082)
+
+Example:
+```java
+definitions.put(
+  "retry_policy_1_codes",
+  ImmutableSet.copyOf(
+      Lists.<StatusCode.Code>newArrayList(
+          StatusCode.Code.DEADLINE_EXCEEDED, StatusCode.Code.UNAVAILABLE)));
+```
+
+Retry parameters are configured [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1086-L1155)
+
+Example:
+```java
+settings =
+  RetrySettings.newBuilder()
+      .setInitialRpcTimeout(Duration.ofMillis(60000L))
+      .setRpcTimeoutMultiplier(1.0)
+      .setMaxRpcTimeout(Duration.ofMillis(60000L))
+      .setTotalTimeout(Duration.ofMillis(60000L))
+      .build();
+```
+
+The configurations above are set [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1306-L1474)
+
+Example:
+```java
+builder
+  .exportAssetsSettings()
+  .setRetryableCodes(RETRYABLE_CODE_DEFINITIONS.get("no_retry_0_codes"))
+  .setRetrySettings(RETRY_PARAM_DEFINITIONS.get("no_retry_0_params"));
```

Comment:
I'm not understanding how this information is to be used by a developer. Is this section just FYI on how the internals are implemented, or is this information important because developers will need to modify this when X.

(The examples below don't seem to require knowledge of the `RETRYABLE_CODE_DEFINITIONS` and `RETRY_PARAM_DEFINITIONS` collections.)

Reply:
I intended this section to be to show where users can find the default retry configurations. I'll update this section to be clearer.

> (The examples below don't seem to require knowledge of the RETRYABLE_CODE_DEFINITIONS and RETRY_PARAM_DEFINITIONS collections.)

Those definitions aren't important and are internal details to gax.